### DISCUSSION TITLE
Can't extract name of block reference

### DISCUSSION THREAD

**Discussion by jamesduke488** [mozman/ezdxf#1109] [Issues]
I'm trying to get the name of block INSERT entities however some of them are returning *UXX instead of their actual name in AutoCAD. I understand these are anonymous block entries and therefore it's a bit more difficult to access some of their info directly but I haven't been able to find any clear workaround either online or from AI prompts.
The blocks have no effective name attributes and I am able to get the correct names for some block references. Any help would be greatly appreciated

**Comment by mozman** [comment]
I can't help much.

`*Uxx` is the name of the block. CAD applications use these anonymous blocks to display the graphical output of some commands such as DIMENSION `*Dxx`, TABLE `*Txx` or ARRAY `*Uxx`.

The only way to get more information about such a block is to look at the DXF structure. The `ezdxf browse` command can help with this. It even has a search function to find the names you are looking for.

**Reply by jamesduke488** [reply]
Yeah, I've looked into the xdata and found that there are references to the original non-dynamic blocks in the xdata of the dynamic blocks. 
My issue now however is that when I try to get the xdata tags through code it's saying that none of the entities have xdata. 
I've tried to check using the individual block appid's (where I've gotten a DXFValueError despite checking spelling and case sensitivity) as well as looping through the entities and using the .xdata attribute. However in both cases it's saying there's no xdata even though there clearly is. 
I'm really confused. If I can just access the xdata of the dynamic blocks I should be able to get the name of the original block

**Comment by mozman** [comment]
Show me your code with the attached zipped DXF file and maybe I can help.

**Comment by mozman** [comment]
https://github.com/mozman/ezdxf/blob/master/src/ezdxf/dynblkhelper.py

This is a helper module based on a previous discussion about dynamic blocks. Keep in mind the dynamic block structure is not documented by Autodesk.

**Comment by jamesduke488** [comment]
Aw man you are an absolute life saver. As soon as I tried it with the
.block_record attribute I was able to get the xdata.

Thank you so much.
